src/target/stm32h7.c

```diff
@@ -264,7 +264,8 @@ bool stm32h7_probe(target_s *target)
 	/* By now it's established that this is likely an H7, but check that it's not an MP15x_CM4 with an errata in AP part code */
 	const uint32_t idcode = target_mem32_read32(target, STM32H7_DBGMCU_IDCODE);
 	const uint16_t dev_id = idcode & STM32H7_DBGMCU_IDCODE_DEV_MASK;
-	DEBUG_TARGET("%s: looking at device ID 0x%03x at 0x%08" PRIx32 "\n", __func__, dev_id, STM32H7_DBGMCU_IDCODE);
+	DEBUG_TARGET(
+		"%s: looking at device ID 0x%03x at 0x%08" PRIu32 "\n", __func__, dev_id, (uint32_t)STM32H7_DBGMCU_IDCODE);
```

Comment:
Why PRIu32? This now prints 0x5c001000 as "0x1543507968", both in BMF and BMDA.

Reply:
Oops, good catch. I created #1889 to fix this.

Reply:
Great catch - thanks ALTracer!